controllers/discordactions.js

```diff
@@ -203,27 +202,33 @@ const updateDiscordNicknames = async (req, res) => {
 
     const discordServerUsers = await dataAccess.retrieveDiscordUsers();
     const nonSuperUsers = discordServerUsers.filter((user) => !user.roles.super_user);
+
     const errorsArr = [];
     let successCounter = 0;
     let errorCounter = 0;
-    const batchUpdate = nonSuperUsers.map(async (user) => {
-      const { discordId, username } = user;
+
+    let counter = 0;
+    for (let i = 0; i < nonSuperUsers.length; i++) {
+      const { discordId, username } = nonSuperUsers[i];
       try {
-        await setUserDiscordNickname(username, discordId);
+        if (counter % 10 === 0 && counter !== 0) {
+          await new Promise((resolve) => setTimeout(resolve, 4500));
+        }
+        await setUserDiscordNickname(username.toLowerCase(), discordId);
+        counter++;
```

Comment:
So if it allows 10 requests at once, then why not do 10 requests all at once and then wait for 3s and then next 10 together instead of waiting for each one?